treeherder/log_parser/utils.py

```diff
@@ -57,11 +48,16 @@ def _retry(e):
         # .retry() raises a RetryTaskError exception,
         # so nothing after this function will be executed
 
-    log_obj = expand_log_url(project, job_guid, job_log_url)
     log_description = "%s %s (%s)" % (project, job_guid, job_log_url)
-
     logger.debug("Downloading/parsing log for %s", log_description)
 
+    job_log = JobLog.objects.get(job__guid=job_guid,
+                                 url=job_log_url)
+
+    if job_log.status == JobLog.PARSED:
```

Comment:
Yes, it is. I had earlier taken out parser_task but then decided to leave it in (partly because I didn't want to change too much at once).

Agreed that there is some inefficiency and redundancy in this code that would be nice to clean up at some point.